src/ferc_eia_match/inputs.py

```diff
@@ -66,9 +67,16 @@ def __init__(
                 and all plant parts will be included in the EIA data.
         """
         self.pudl_out = pudl_out
+        self.pudl_out.freq = "AS"
         # TODO: need linking_tool?
         self.linking_tool = linking_tool
+        # company name string cleaner, currently uses default rules
+        self.utility_cleaner = financial_entity_cleaner.text.name.CompanyNameCleaner()
         self.report_years = report_years
+        self.plant_parts_eia = None
+        if self.report_years:
+            self.pudl_out.start_date = str(min(self.report_years)) + "-01-01"
+            self.pudl_out.end_date = str(max(self.report_years)) + "-12-31"
```

Comment:
I think forcibly resetting these attributes has a similar issue as resetting the frequency above.  Either we should be creating the `pudl_out` as we expect it to exist here inside the method, or we should be checking that these attributes are as we need them to be. Mutating attributes of the object that could make its internal state internally inconsistent could easily get very messy.